src/main/java/roomescape/dao/TimeDao.java

```diff
@@ -0,0 +1,55 @@
+package roomescape.dao;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.domain.Time;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+
+@Repository
+public class TimeDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    public TimeDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<Time> timeRowMapper = (resultSet, rowNum) ->
+            new Time(
+                    resultSet.getLong("id"),
+                    resultSet.getString("time")
+            );
+
+    public Time createTime(String time) {
+        String sql = "INSERT INTO time (time) VALUES (?)";
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, new String[]{"id"});
+            ps.setString(1, time);
+            return ps;
+        }, keyHolder);
+
+        Long id = keyHolder.getKey().longValue();
+        return new Time(id, time);
+    }
+
+    public List<Time> findAllTimes() {
+        String sql = "SELECT id, time FROM time";
+        return jdbcTemplate.query(sql, timeRowMapper);
+    }
+
+    public Time findById(Long id) {
+        String sql = "SELECT id, time FROM time WHERE id = ?";
+        return jdbcTemplate.queryForObject(sql, timeRowMapper, id);
+    }
+
+    public void deleteTime(Long id) {
+        String sql = "DELETE FROM time WHERE id = ?";
+        jdbcTemplate.update(sql, id);
+    }
```

Comment:
```suggestion
    public Time findById(Long id) {
        String sql = "SELECT id, time FROM time WHERE id = ?";
        return jdbcTemplate.queryForObject(sql, timeRowMapper, id);
    }

    public void deleteById(Long id) {
        String sql = "DELETE FROM time WHERE id = ?";
        jdbcTemplate.update(sql, id);
    }
```

- findById와 같이 delete 메서드도 같은 패턴으로 메서드명을 작성하면 더 좋을 것 같아요!
- 추가로 저는 `deleteTime()`과 같이 Time을 추가하는 것을 선호하지는 않아요. 왜냐하면 결국 해당 클래스(TimeDao)가 Time 도메인에 대한 작업이 수행된다는 것을 알 수 있기 때문이에요.
```java
// 예시를 들기 위해 직접 생성합니다.
TimeDao timeDao = new TimeDao();

timeDao.deleteTime(1); // 기존 메서드명은 변수와 메서드에서 중복된 코드(time이라는!)가 발생된다고 생각해요.
timeDao.deleteById(1); // 변수로 어떤 도메인에 접근하는지 파악할 수 있고 id로 삭제한다고 명확히 표현할 수 있다고 생각해요.
```

Reply:
오 그렇네요. 어차피 코드 상으로 무엇을 삭제하는 건지는 아니까 굳이 메서드명에서 보여줄 필요는 없었는데 실수한거같아용. 감사합니다